From 0fc6d304fab8c1131a977fcb8200c485f4bc9f98 Mon Sep 17 00:00:00 2001
From: Xiao Chen <xiao@apache.org>
Date: Thu, 8 Jun 2017 22:35:31 -0700
Subject: [PATCH 2507/2540] HADOOP-13854. KMS should log error details in
 KMSExceptionsProvider.

(cherry picked from commit c4b5c32669423b9a792f33f9f8333d95528f2515)
(cherry picked from commit aaf533e8a06b0f5b9825c90eeb1cb3844580a690)

Change-Id: I61278da53933a43967026829775b78274be38fd9
(cherry picked from commit 539a4ba0d90af7f31ee7a398d313ec6593833cdf)
---
 .../apache/hadoop/crypto/key/kms/server/KMS.java   |    2 +-
 .../key/kms/server/KMSExceptionsProvider.java      |    4 ++++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/hadoop-common-project/hadoop-kms/src/main/java/org/apache/hadoop/crypto/key/kms/server/KMS.java b/hadoop-common-project/hadoop-kms/src/main/java/org/apache/hadoop/crypto/key/kms/server/KMS.java
index d8755ec..e9d6a7a 100644
--- a/hadoop-common-project/hadoop-kms/src/main/java/org/apache/hadoop/crypto/key/kms/server/KMS.java
+++ b/hadoop-common-project/hadoop-kms/src/main/java/org/apache/hadoop/crypto/key/kms/server/KMS.java
@@ -70,7 +70,7 @@
   private KeyProviderCryptoExtension provider;
   private KMSAudit kmsAudit;
 
-  private static final Logger LOG = LoggerFactory.getLogger(KMS.class);
+  static final Logger LOG = LoggerFactory.getLogger(KMS.class);
 
   public KMS() throws Exception {
     provider = KMSWebApp.getKeyProvider();
diff --git a/hadoop-common-project/hadoop-kms/src/main/java/org/apache/hadoop/crypto/key/kms/server/KMSExceptionsProvider.java b/hadoop-common-project/hadoop-kms/src/main/java/org/apache/hadoop/crypto/key/kms/server/KMSExceptionsProvider.java
index 5cb0885..72e0c9a 100644
--- a/hadoop-common-project/hadoop-kms/src/main/java/org/apache/hadoop/crypto/key/kms/server/KMSExceptionsProvider.java
+++ b/hadoop-common-project/hadoop-kms/src/main/java/org/apache/hadoop/crypto/key/kms/server/KMSExceptionsProvider.java
@@ -43,6 +43,7 @@
 public class KMSExceptionsProvider implements ExceptionMapper<Exception> {
   private static Logger LOG =
       LoggerFactory.getLogger(KMSExceptionsProvider.class);
+  private final static Logger EXCEPTION_LOG = KMS.LOG;
 
   private static final String ENTER = System.getProperty("line.separator");
 
@@ -98,6 +99,9 @@ public Response toResponse(Exception exception) {
           KMSMDCFilter.getMethod(),
           KMSMDCFilter.getURL(), getOneLineMessage(exception));
     }
+    EXCEPTION_LOG.warn("User {} request {} {} caused exception.",
+        KMSMDCFilter.getUgi(), KMSMDCFilter.getMethod(),
+        KMSMDCFilter.getURL(), exception);
     return createResponse(status, throwable);
   }
 
-- 
1.7.9.5

